### PR TITLE
Memory pair game

### PR DESCRIPTION
**Pull request by TaniaBondarenko**
Memory – Pair Game

[Demo](https://taniabondarenko.github.io/memory-pair-game/)
[Code base](https://github.com/TaniaBondarenko/memory-pair-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by andrewklmn** [comment]
Nice job!
The look of your game is very actual for this time. )))

I found some issues with playing this game.
If I quickly click more than two cards they can disappear together like this: 

Peek 2021-01-11 12-21

It may be better to implement a check of the number of opened cards. Just for preventing opening the third one.
This is not a code review. Just wondering.

File: submissions/TaniaBondarenko/memory-pair-game/app.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    currentIndex -= 1;
+    temporaryValue = arr[currentIndex];
+    arr[currentIndex] = arr[randomIndex];
+    arr[randomIndex] = temporaryValue;
+  };
+};
+
+setTimeout(() => {
+    const CARDS = Array.from(document.querySelectorAll(".card_inner"));
+    CARDS.forEach(card => card.addEventListener('click', flipCard));
```
Comment (zonzujiro) [inline submissions/TaniaBondarenko/memory-pair-game/app.js:62]: Event delegation please

File: submissions/TaniaBondarenko/memory-pair-game/app.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+}, DELAY1);
```
Comment (zonzujiro) [inline submissions/TaniaBondarenko/memory-pair-game/app.js:63]: Magic number in name - it is bad. Don't use magic numbers

File: submissions/TaniaBondarenko/memory-pair-game/app.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+function flipCard(e) {
+  const CARD_INNER = e.target.parentElement;
```
Comment (zonzujiro) [inline submissions/TaniaBondarenko/memory-pair-game/app.js:66]: No need in uppercase - it is not global constant.

File: submissions/TaniaBondarenko/memory-pair-game/app.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  CARD_INNER.classList.add('is-flipped');
+  cardsFlipped.push(CARD_INNER); 
+  if (cardsFlipped.length === 2) {
+    isMatch();
+  } else {
+    cardsFlipped;
```
Comment (zonzujiro) [inline submissions/TaniaBondarenko/memory-pair-game/app.js:72]: Ehhhhm? What it means and what it should do?

File: submissions/TaniaBondarenko/memory-pair-game/app.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  };
+};
+
+function isMatch() {
+  numOfAttempts++;
```
Comment (zonzujiro) [inline submissions/TaniaBondarenko/memory-pair-game/app.js:77]: `const [firstCard, secondCard] = cardsFlipped;` - better then `[0]`

File: submissions/TaniaBondarenko/memory-pair-game/app.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  cardsFlipped = [];
+};
+
+function wonGame() {
+  if (pairsOpened === CARDS_LENGTH / 2) {
+    setTimeout(() => {
+      askPlayer();
+    }, DELAY1);
+  } else {
+    return;
```
Comment (zonzujiro) [inline submissions/TaniaBondarenko/memory-pair-game/app.js:111]: No need

File: submissions/TaniaBondarenko/memory-pair-game/app.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        window.removeEventListener("load", createBoard);
+        mode.removeEventListener("click", changeMode);
+      };
+};
+
+let mode = document.querySelector(".mode");
+mode.addEventListener("click", changeMode);
+
+function changeMode() {
+  let bodyBack = document.querySelector(".background");
```
Comment (zonzujiro) [inline submissions/TaniaBondarenko/memory-pair-game/app.js:129]: Redundant variable, IMO

**Review by zonzujiro — CHANGES_REQUESTED** [review]

File: submissions/TaniaBondarenko/memory-pair-game/app.js (hunk 3 above)
Comment (TaniaBondarenko) [reply submissions/TaniaBondarenko/memory-pair-game/app.js:66]: Got it

File: submissions/TaniaBondarenko/memory-pair-game/app.js (hunk 2 above)
Comment (TaniaBondarenko) [reply submissions/TaniaBondarenko/memory-pair-game/app.js:63]: got it

File: submissions/TaniaBondarenko/memory-pair-game/app.js (hunk 7 above)
Comment (TaniaBondarenko) [reply submissions/TaniaBondarenko/memory-pair-game/app.js:129]: got it

File: submissions/TaniaBondarenko/memory-pair-game/app.js (hunk 6 above)
Comment (TaniaBondarenko) [reply submissions/TaniaBondarenko/memory-pair-game/app.js:111]: Checked. Okay

File: submissions/TaniaBondarenko/memory-pair-game/app.js (hunk 4 above)
Comment (TaniaBondarenko) [reply submissions/TaniaBondarenko/memory-pair-game/app.js:72]: It should return array of cardsFlippes with less than length=2 to continue working.
But I'll check this more carefully.

File: submissions/TaniaBondarenko/memory-pair-game/app.js (hunk 1 above)
Comment (TaniaBondarenko) [reply submissions/TaniaBondarenko/memory-pair-game/app.js:62]: Okay

File: submissions/TaniaBondarenko/memory-pair-game/app.js (hunk 4 above)
Comment (zonzujiro) [reply submissions/TaniaBondarenko/memory-pair-game/app.js:72]: I don't see `return` here :)

Comment (TaniaBondarenko) [reply submissions/TaniaBondarenko/memory-pair-game/app.js:72]: Yes, rewritten it.

File: submissions/TaniaBondarenko/memory-pair-game/app.js (hunk 5 above)
Comment (TaniaBondarenko) [reply submissions/TaniaBondarenko/memory-pair-game/app.js:77]: I used your prompt and realized, that I don't need such array at all.

**Comment by TaniaBondarenko** [comment]
@andrewklmn, thank you! I think I fixed that bug.
@zonzujiro, updated! Please check.

**Pull request merged**